scene/main/viewport.cpp

```diff
@@ -3612,9 +3625,31 @@ void Viewport::_propagate_exit_world_3d(Node *p_node) {
 }
 
 void Viewport::set_use_xr(bool p_use_xr) {
-	use_xr = p_use_xr;
-
-	RS::get_singleton()->viewport_set_use_xr(viewport, use_xr);
+	if (use_xr != p_use_xr) {
+		use_xr = p_use_xr;
+
+		RS::get_singleton()->viewport_set_use_xr(viewport, use_xr);
+
+		if (use_xr) {
+			pre_xr_size = size;
```

Comment:
I'm not a fan of recording the pre_xr_size. IMHO the logic here should be that when `use_xr` becomes false we leave the size as is for normal viewports, we do reset the viewcount to 1. There is even an argument for setting the size to 0,0. It is up the the user to decide what they want the viewport to do when it is no longer an XR viewport.

We can make an exception for the root viewport and reset the size to the screensize. We should be able to obtain this from the OS singleton.

Reply:
Ok, sounds good!